In [1]:
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
import numpy as np

# 假設真實值和預測值
y_true = np.array([100, 200, 300, 400, 500])
y_pred = np.array([110, 190, 320, 380, 510])

mse  = mean_squared_error(y_true, y_pred)
rmse = np.sqrt(mse)
mae  = mean_absolute_error(y_true, y_pred)
r2   = r2_score(y_true, y_pred)

print(f"MSE:  {mse:.2f}")
print(f"RMSE: {rmse:.2f}")
print(f"MAE:  {mae:.2f}")
print(f"R²:   {r2:.4f}")

MSE:  220.00
RMSE: 14.83
MAE:  14.00
R²:   0.9890


MAE  = 14    → 平均每筆預測差 14 
RMSE = 14.83 → 跟 MAE 接近，代表沒有很離譜的異常預測
MSE  = 220   → 單位是元²，不直覺，所以才開根號變 RMSE
R²   = 0.989 → 解釋了 98.9% 的變化，非常好